### PR TITLE
Fix #242 - Warn about unused labels

### LINKED ISSUE #242: Warn about unused labels
DMD enhancement: https://issues.dlang.org/show_bug.cgi?id=6449

### PR DESCRIPTION
**Pull request by WalterWaldron**

### CODE REVIEW

File: src/analysis/unused_label.d [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+class UnusedLabelCheck : BaseAnalyzer
+{
+	alias visit = BaseAnalyzer.visit;
+
+	this(string fileName)
+	{
+		super(fileName);
+	}
+
+	struct Label
```
Comment (Hackerpilot) [inline src/analysis/unused_label.d]: This should be `static struct`.

File: src/analysis/unused_label.d [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+			label.column = token.column;
+		}
+		labeledStatement.declarationOrStatement.accept(this);
+	}
+
+	void labelUsed(string name)
+	{
+		Label* entry = name in current;
+		if (entry is null)
+			current[name] = Label(name, -1, -1, true);
```
Comment (Hackerpilot) [inline src/analysis/unused_label.d]: I'm confused by the use of -1 here. Did you mean `size_t.max`?

File: src/analysis/unused_label.d [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			F:
+				return;
+			}();
+		F: // [warn]: Label "F" is not used.
+			return x;
+		}
+	}c, sac);
+
+	stderr.writeln("Unittest for UnusedLabelCheck passed.");
+}
```
Comment (Hackerpilot) [inline src/analysis/unused_label.d]: Missing a newline.

File: src/analysis/unused_label.d (hunk 2 above)
Comment (WalterWaldron) [inline src/analysis/unused_label.d]: Yes, I used that as shorthand for ```size_t.max```. I've updated the PR.

**Comment by Hackerpilot** [comment]
Thanks for implementing this.

**Comment by BBasile** [comment]
just found this bug when testing the issue:

```D
void main(string[] args)
{
    LBL:
}
```

crashes dscanner. (dscanner file.d --s)

**Comment by Hackerpilot** [comment]
Can you file that as a separate issue? It's easy for it to get lost if it's just a comment an a merge request.

**Pull request merged**